src/compressed_chunk.c

```diff
@@ -33,6 +42,129 @@ void Compressed_FreeChunk(Chunk_t *chunk) {
     free(chunk);
 }
 
+static void swapChunks(CompressedChunk *a, CompressedChunk *b) {
+    CompressedChunk tmp = *a;
+    *a = *b;
+    *b = tmp;
+}
+
+static void extendChunk(CompressedChunk *chunk) {
+    chunk->size += 64;
+    chunk->data = (u_int64_t *)realloc(chunk->data, chunk->size * sizeof(char));
+}
+
+static void trimChunk(CompressedChunk *chunk) {
+    int excess = chunk->size - (chunk->idx + BIT) / BIT;
+
+    assert(excess >= 0); // else we have written beyond allocated memory
+
+    if (excess > 0) {
+        size_t newSize = chunk->size - excess;
+        chunk->data = realloc(chunk->data, newSize);
+        chunk->size = newSize;
+    }
+}
+
+Chunk_t *Compressed_SplitChunk(Chunk_t *chunk) {
+    CompressedChunk *curChunk = chunk;
+    size_t split = curChunk->count / 2;
+    size_t curNumSamples = curChunk->count - split;
+
+    // add samples in new chunks
+    size_t i = 0;
+    Sample sample;
+    ChunkIter_t *iter = Compressed_NewChunkIterator(curChunk, false);
+    CompressedChunk *newChunk1 = Compressed_NewChunk(curChunk->size / sizeof(Sample));
+    CompressedChunk *newChunk2 = Compressed_NewChunk(curChunk->size / sizeof(Sample));
+    for (; i < curNumSamples; ++i) {
+        Compressed_ChunkIteratorGetNext(iter, &sample);
+        if (Compressed_AddSample(newChunk1, &sample) != CR_OK) {
+            goto error;
```

Comment:
lets try to rewrite without gotos, unstructured programming can lead to funky bugs and issues in the future.